cmd/openshift-install/targets.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/cluster"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/kubeconfig"
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/metadata"
+)
+
+var (
+	installConfigCmd = &cobra.Command{
+		Use:   "install-config",
+		Short: "Generates the Install Config asset",
+		Long:  "",
+		RunE:  runTargetCmd(&installconfig.InstallConfig{}),
+	}
+
+	ignitionConfigsCmd = &cobra.Command{
+		Use:   "ignition-configs",
+		Short: "Generates the Ignition Config asset",
+		Long:  "",
+		RunE:  runTargetCmd(&bootstrap.Bootstrap{}, &machine.Master{}, &machine.Worker{}),
+	}
+
+	manifestsCmd = &cobra.Command{
+		Use:   "manifests",
+		Short: "Generates the Kubernetes manifests",
+		Long:  "",
+		RunE:  runTargetCmd(&manifests.Manifests{}, &manifests.Tectonic{}),
+	}
+
+	clusterCmd = &cobra.Command{
+		Use:   "cluster",
+		Short: "Create an OpenShift cluster",
+		Long:  "",
+		RunE:  runTargetCmd(&cluster.TerraformVariables{}, &kubeconfig.Admin{}, &cluster.Cluster{}, &metadata.Metadata{}),
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(installConfigCmd)
```

Comment:
I'd rather have these in `main.go` like [this][1].

[1]: https://github.com/kubernetes/kubernetes/blob/v1.12.1/pkg/kubectl/cmd/cmd.go#L404

Reply:
i think `kubectl` example is more useful if we have more complex commands. but i can add `new<cmd>` type funcs.

Reply:
No I just meant "not in an `init()`" ;).